src/jon_mikkel_ex/ex05/myrand.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+
+__author__ = "Jon-Mikkel Korsvik"
+__email__ = "jonkors@nmbu.no"
+
+
+class LCGRand:
+    slope = 7 ** 5
+    congruence_class = 2 ** 31 - 1
+
+    def __init__(self, seed):
+        """
+
+        :param seed: A number selected when initializing, integer, set into
+        a list.
+        a: is a large odd number
+        m: is a large odd number
+        """
+        self.hidden = seed
+
+    def rand(self):
+        """
+        Index (idx) is updated each time rand is called. The seed list is
+        appended with the newly calculated "random number" based on the
+        first seed.
+        :return: the last seed from the list. list of integers
+        """
+        self.hidden *= self.slope
+        self.hidden %= self.congruence_class
+        return self.hidden
+
+    def random_sequence(self, length):
+        return RandIter(self, length)
+
+    def infinite_random_sequence(self):
+        """
+        Generate an infinite sequence of random numbers.
+
+        Yields
+        ------
+        int
+            A random number.
+        """
+        while True:
+            yield self.rand()
+        # Alternatively you could:
+        # return RandIter(self, 0)
+        # The code can handle both ways
```

Comment:
If that's the case, `RandIter` doesn't really work as it should. I suggest putting `if self.num_generated_numbers >= self.length` in line `106` to hopefully avoid this.

Reply:
alternatively python (or numpy, i don't recall) has an `inf`class. which will always compare grater than another value which could be used as argument to achieve this.